### PR TITLE
Add Agile flag

### PR DESCRIPTION
**Pull request by pmpurifoy**
Fixed issue with activation on STA thread by adding REGCLS_AGILE.

### CODE REVIEW

**Comment by David-Quero** [comment]
Looks good to me, how did you validate it?

**Comment by loneursid** [comment]
Did you run it against the winui sample or did you use something else?

**Comment by pmpurifoy** [comment]
Validation was done using the inner loop demo app for console. WinUI verification was done by creating a nuget packagewith this change and verifying GetActivatedEventArgs succeeds in OnLaunched for AppNotification activation scenario.

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 1]
```diff
@@ -468,12 +468,15 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                 {
                     auto lock{ m_lock.lock_exclusive() };
                     // Register a PushNotificationBackgroundTask to handle background activation scenarios
+
```
Comment (David-Quero) [inline dev/PushNotifications/PushNotificationManager.cpp:471]: I think this blank line is in the wrong place

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 2]
```diff
@@ -139,11 +139,12 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 
                 // Check if the caller has registered event handlers, if so the REGCLS_MULTIPLEUSE flag will cause COM to ensure that all activators
                 // are routed inproc, otherwise with REGCLS_MULTIPLEUSE COM will launch a new process of the Win32 app for each invocation.
+                auto activationFlag{ m_notificationHandlers ? REGCLS_MULTIPLEUSE : REGCLS_SINGLEUSE };
                 THROW_IF_FAILED(::CoRegisterClassObject(
                     AppModel::Identity::IsPackagedProcess() ? registeredClsid : storedComActivatorGuid,
                     winrt::make<AppNotificationManagerFactory>().get(),
                     CLSCTX_LOCAL_SERVER,
-                    m_notificationHandlers ? REGCLS_MULTIPLEUSE : REGCLS_SINGLEUSE,
+                    activationFlag | REGCLS_AGILE,
```
Comment (hulumane) [inline dev/AppNotifications/AppNotificationManager.cpp:147]: Needs a comment as to why the AGILE flag is added. It's non standard.

Something like:
// In WinUI Application::OnLaunch, the thread context in which CoRegisterClassObject can be an STA. So a subsequent invoke on that object from an MTA can lead to an error unless the object is declared as agile (neutral apartment)

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 3]
```diff
@@ -468,12 +468,15 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
...
                     m_waitHandleForArgs.create();
+
+                    auto activationFlags{ m_foregroundHandlers ? REGCLS_MULTIPLEUSE : REGCLS_SINGLEUSE };
                     THROW_IF_FAILED(::CoRegisterClassObject(
                         m_registeredClsid,
                         winrt::make<PushNotificationManagerFactory>().get(),
                         CLSCTX_LOCAL_SERVER,
-                        m_foregroundHandlers ? REGCLS_MULTIPLEUSE : REGCLS_SINGLEUSE,
+                        activationFlags | REGCLS_AGILE,
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationManager.cpp:479]: Comment needed here too

**Review by loneursid — APPROVED** [review]
Verified using the C++ WinUI Unpackaged sample and this does indeed fix the issue

**Review by hulumane — APPROVED** [review]

**Pull request merged**